### PR TITLE
XWIKI-12629: Page Right is shown when administering inexistent WebHome page

### PR DESCRIPTION
**Pull request by Sereza7**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-12629

# Changes

## Description

* Fixed test that relied on an uncreated page's administration section

## Clarifications

* This is a fix for a regression introduced in https://github.com/xwiki/xwiki-platform/pull/3590 : https://ci.xwiki.org/job/XWiki/job/xwiki-platform/job/master/7578/testReport/org.xwiki.administration.test.ui/AllIT$NestedAdministrationIT/Platform_Builds___main___integration_tests___IT_for_xwiki_platform_core_xwiki_platform_administration_xwiki_platform_administration_test_xwiki_platform_administration_test_docker___Build_for_IT_for_xwiki_platform_core_xwiki_platform_administration_xwiki_platform_administration_test_xwiki_platform_administration_test_docker___verifyAdministrationSections_TestUtils_/ 
* There was a comment about not creating the page for performance. Since editing the administration section creates the page in the background, I'm pretty sure there is close to none performance improvement. 
* This behaviour of the administration section was considered a bug, so I updated the test to create a page and then use its administration section regularly.

# Screenshots & Video

None, test only change

# Executed Tests

Successfully ran `mvn clean install -f xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker -Pdocker`

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * 16.4.X
  * 15.10.X
Similar to https://github.com/xwiki/xwiki-platform/pull/3590

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/AdministrationIT.java [hunk 1]
```diff
@@ -71,11 +72,11 @@ void verifyAdministrationSections(TestUtils setup)
         assertTrue(wikiAdministrationPage.hasNotSection("PageRights"));
 
         // Select XWiki page administration.
-        setup.gotoPage("NonExistentSpace", "WebHome");
+        setup.gotoPage(testReference);
         page = new AdministrablePage();
         AdministrationPage pageAdministrationPage = page.clickAdministerPage();
 
-        assertEquals("Page Administration: NonExistentSpace", pageAdministrationPage.getDocumentTitle());
+        assertEquals("Page Administration: AdministrationIT.verifyAdministrationSections", pageAdministrationPage.getDocumentTitle());
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/AdministrationIT.java:79]: This won't work if you execute the whole test suite as the test class will be NestedAdministrationIT. You need to use a serialization of the testReference.

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/AdministrationIT.java]: Addressed in https://github.com/xwiki/xwiki-platform/pull/3603/commits/61c26755c85b3a8b077e4eff4caea25e9799360a . I tested with the singular test already, but I'm currently running the whole test suite. I'll update this if the whole test suite does not pass somehow.

File: xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/AdministrationIT.java [hunk 2]
```diff
@@ -71,11 +72,12 @@ void verifyAdministrationSections(TestUtils setup)
         assertTrue(wikiAdministrationPage.hasNotSection("PageRights"));
 
         // Select XWiki page administration.
-        setup.gotoPage("NonExistentSpace", "WebHome");
+        setup.gotoPage(testReference);
         page = new AdministrablePage();
         AdministrationPage pageAdministrationPage = page.clickAdministerPage();
-
-        assertEquals("Page Administration: NonExistentSpace", pageAdministrationPage.getDocumentTitle());
+        String fullName = setup.serializeReference(testReference.getParent()).split(":")[1];
```
Comment (tmortagne) [inline xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/AdministrationIT.java:78]: I think what you were searching for is #serializeLocalReference (which does not contain the wiki part).

Comment (surli) [reply xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/AdministrationIT.java]: Hmmm I didn't paid enough attention: that one is not available in 15.10.x / 16.4.x. Backporting it.

**Pull request merged**